File: online-judge-solutions/Codeforces/78A.cpp
//https://codeforces.com/problemset/problem/78/A
//Lang		:	GNU C++14
//Time		:	30 ms
//Memory	:	0 KB
#include<iostream>
using namespace std;
int main(){
	int n=3,a[3]={5,7,5};
	bool r=true;
	while(n--){
		string s;getline(cin,s);
		int vocales=0;
		for(char x:s){
			if(x=='a' || x=='e' || x=='i' || x=='o' || x=='u') vocales++;
		}
		if(vocales!=a[n]) r=false;
	}
	cout<<(r?"YES":"NO")<<"\n";
	return 0;
}

File: online-judge-solutions/Codeforces/78A.py
#https://codeforces.com/problemset/problem/78/A
#Lang		:	Python 3.8
#Time		:	92 ms
#Memory         :	0 KB
a,r=[5,7,5],True
for z in range(3):
    s=input()
    vocales=0
    for i in range(len(s)):
        if(s[i]=='a' or s[i]=='e' or s[i]=='i' or s[i]=='o' or s[i]=='u'): vocales+=1
    if(vocales!=a[z]): r=False
print("YES" if r else "NO")
